shared/filestore/s3store.go

```diff
@@ -148,19 +161,19 @@ func (b *S3FileBackend) TestConnection() error {
 		if obj.Err != nil {
 			typedErr := s3.ToErrorResponse(obj.Err)
 			if typedErr.Code != bucketNotFound {
-				return errors.Wrap(err, "unable to list objects in the s3 bucket")
+				return &S3FileBackendAuthError{DetailedError: "unable to list objects in the S3 bucket"}
 			}
 			exists = false
 		}
 	} else {
 		exists, err = b.client.BucketExists(context.Background(), b.bucket)
 		if err != nil {
-			return errors.Wrap(err, "unable to check if the s3 bucket exists")
+			return &S3FileBackendAuthError{DetailedError: "unable to check if the S3 bucket exists"}
 		}
 	}
 
 	if !exists {
-		return ErrNoS3Bucket
+		return &S3FileBackendNoBucketError{}
```

Comment:
Are we not able to define this in one line with `errors.New("no such bucket")`?

Reply:
As mentioned above, I use a type switch in `app/file.go` to determine which error message to display to the user.